linetools/guis/xspecgui.py

```diff
@@ -75,6 +75,18 @@ def __init__(self, ispec, parent=None, zsys=None, norm=None, exten=None,
                                                 zsys=zsys, norm=norm, exten=exten,
                                                 screen_scale=self.scale,
                                                  rsp_kwargs=rsp_kwargs, **kwargs)
+        # Reset redshift from spec
+        if zsys is None:
+            if hasattr(self.spec_widg.spec, 'z'):
+                self.pltline_widg.setz(str(self.spec_widg.spec.z[self.spec_widg.select]))
```

Comment:
I tried this feature out with spectra from a specdb.spectra_from_coord() query, and the first spectrum has its redshift set, but the redshift does not change when I select a different spectrum.  Can the multiSpecWidget talk back to the main gui to change redshift when selecting different spectra?

Reply:
fixed;  great catch